acl/acl.go

```diff
@@ -433,6 +441,33 @@ func (p *PolicyACL) KeyRead(key string) bool {
 	return p.parent.KeyRead(key)
 }
 
+// KeyWritePrefix returns if a prefix is allowed to be read
+func (p *PolicyACL) KeyReadPrefix(prefix string) bool {
+	// Look for a matching rule
+	_, rule, ok := p.keyRules.LongestPrefix(prefix)
+
+	// Look if any of our children have a read policy
+	subRuleFound := false
+	p.keyRules.WalkPrefix(prefix, func(path string, rule interface{}) bool {
+		if rule.(string) == PolicyRead || rule.(string) == PolicyWrite {
+			subRuleFound = true
```

Comment:
This also means "and can read" so I'd maybe name this `canRead` or something like that.